src/itdelatrisu/opsu/audio/MultiClip.java

```diff
@@ -186,10 +195,17 @@ private Clip getClip() throws LineUnavailableException {
 		} else {
 			// create a new clip
 			// NOTE: AudioSystem.getClip() doesn't work on some Linux setups.
+			Log.debug("creating a new clip");
 			DataLine.Info info = new DataLine.Info(Clip.class, format);
 			c = (Clip) AudioSystem.getLine(info);
 			if (format != null)
 				c.open(format, audioData, 0, audioData.length);
+			// This is a little hacky, but we can't do an instanceof check and
+			// there's no reason to add the listener unless the system is using
+			// PulseAudioClip et al.
+			if (c.getClass().getSimpleName().equals("PulseAudioClip")) {
+				c.addLineListener(new PulseAudioFixerListener(c));

```

Comment:
When I first wrote the listener, I had it inlined, but it resulted in a proliferation of threads. I moved the listener to a defined class and switch to a static cached thread pool which works really well at keeping extraneous threads low. I also like keeping the ugliness tied to a particular Clip implementation separate.

That being said, I can move the static cached thread pool to the Multiclip class and inline the listener or I can do something like keep the class declaration and just move it to the MutlipClip file. What do you think?